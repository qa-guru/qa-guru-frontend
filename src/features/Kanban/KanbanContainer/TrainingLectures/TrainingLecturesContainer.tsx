import React, { useContext } from "react";
import { ITrainingLecturesContainer } from "./TrainingLecturesContainer.types";
import { useTrainingLecturesQuery } from "../../../../api/graphql/trainingLecture/trainingLectures";
import LectureSelection from "../../KanbanView/components/Form/LectureSelection";
import { SelectedTrainingIdContext } from "../../../../context/SelectedTrainingIdContext";

const TrainingLecturesContainer: React.FC<ITrainingLecturesContainer> = ({
  control,
}) => {
  const { selectedTrainingId } = useContext(SelectedTrainingIdContext);

  const { data, loading } = useTrainingLecturesQuery({
    variables: { id: selectedTrainingId! },
    skip: !selectedTrainingId,
  });

  return <LectureSelection data={data!} control={control} />;
};

export default TrainingLecturesContainer;
