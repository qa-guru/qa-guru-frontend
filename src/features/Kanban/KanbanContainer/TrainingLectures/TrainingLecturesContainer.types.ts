export interface ITrainingLecturesContainer {
  control: any;
}
