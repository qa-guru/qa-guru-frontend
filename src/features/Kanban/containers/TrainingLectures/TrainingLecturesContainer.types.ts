import { Control } from "react-hook-form";
import { IFilterKanban } from "../../views/Form/Form.types";

export interface ITrainingLecturesContainer {
  control: Control<IFilterKanban, unknown>;
}
