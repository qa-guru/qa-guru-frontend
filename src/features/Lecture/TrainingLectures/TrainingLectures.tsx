import React from "react";
import { CardActionArea, Grid, Paper, Typography } from "@mui/material";
import { Link } from "react-router-dom";
import styles from "./TrainingLectures.module.scss";
import { TrainingLecturesQuery } from "../../../generated/graphql";

interface ITrainingLectures {
  data: TrainingLecturesQuery;
  trainingId: string;
}

const TrainingLectures: React.FC<ITrainingLectures> = ({
  data,
  trainingId,
}) => {
  return (
    <Grid container spacing={2}>
      {data?.trainingLectures?.map((item, index) => {
        return (
          <Grid item xs={12} key={index}>
            <Link
              className={styles.link}
              to={`/training/${trainingId}/${item?.lecture?.id}`}
            >
              <CardActionArea>
                <Paper className={styles.paper}>
                  <Typography variant="h3">{item?.lecture?.subject}</Typography>
                  <Typography variant="subtitle2" mt="16px">
                    {item?.lecture?.description}
                  </Typography>
                  <Typography color="primary" variant="h4" align="right">
                    Продолжить
                  </Typography>
                </Paper>
              </CardActionArea>
            </Link>
          </Grid>
        );
      })}
    </Grid>
  );
};

export default TrainingLectures;
