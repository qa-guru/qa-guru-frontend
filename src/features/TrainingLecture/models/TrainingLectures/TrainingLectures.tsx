import React from "react";
import { Box, CardActionArea, Paper, Typography } from "@mui/material";
import { Link } from "react-router-dom";
import styles from "./TrainingLectures.module.scss";
import { TrainingLecturesQuery } from "../../../../generated/graphql";

interface ITrainingLectures {
  data: TrainingLecturesQuery;
  trainingId: string;
}

const TrainingLectures: React.FC<ITrainingLectures> = ({
  data,
  trainingId,
}) => {
  return (
    <Box className={styles.box}>
      {data?.trainingLectures?.map((item, index) => {
        return (
          <Link
            className={styles.link}
            key={index}
            to={`/training/${trainingId}/${item?.lecture?.id}`}
          >
            <CardActionArea>
              <Paper className={styles.paper}>
                <Typography className={styles.title}>
                  {item?.lecture?.subject}
                </Typography>
                <Typography className={styles.description}>
                  {item?.lecture?.description}
                </Typography>
              </Paper>
            </CardActionArea>
          </Link>
        );
      })}
    </Box>
  );
};

export default TrainingLectures;
