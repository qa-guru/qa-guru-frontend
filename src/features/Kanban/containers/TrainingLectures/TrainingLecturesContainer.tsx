import React, { useContext } from "react";
import { ITrainingLecturesContainer } from "./TrainingLecturesContainer.types";
import { useTrainingLecturesQuery } from "../../../../api/graphql/trainingLecture/trainingLectures";
import LectureSelection from "../../views/Form/LectureSelection";
import { KanbanContext } from "../../context/KanbanContext";

const TrainingLecturesContainer: React.FC<ITrainingLecturesContainer> = ({
  control,
}) => {
  const { selectedTrainingId } = useContext(KanbanContext);

  const { data, loading } = useTrainingLecturesQuery({
    variables: { id: selectedTrainingId! },
    skip: !selectedTrainingId,
  });

  return <LectureSelection data={data!} control={control} />;
};

export default TrainingLecturesContainer;
