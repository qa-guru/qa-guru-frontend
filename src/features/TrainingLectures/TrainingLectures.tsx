import React from "react";
import { CardActionArea, Grid, Paper, Stack, Typography } from "@mui/material";
import { Link } from "react-router-dom";
import { ITrainingLectures } from "./TrainingLectures.types";
import { primary, white } from "../../theme/colors";

const style = {
  link: { textDecoration: "none" },
  paper: { padding: "24px 24px 15px 24px" },
  circle: {
    minWidth: "40px",
    height: "40px",
    bgcolor: primary.main,
    borderRadius: "50%",
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    color: white.main,
  },
};

const TrainingLectures: React.FC<ITrainingLectures> = (props) => {
  const { dataTrainingLectures, trainingId, dataTraining } = props;
  const { trainingLectures } = dataTrainingLectures;
  const { name } = dataTraining.training!;

  return (
    <>
      <Typography mb="20px" variant="h4">
        {name}
      </Typography>
      <Grid container spacing={2}>
        {trainingLectures?.map((item, index) => {
          const { id, subject, description } = item!.lecture!;

          return (
            <Grid item xs={12} key={index}>
              <Link style={style.link} to={`/training/${trainingId}/${id}`}>
                <CardActionArea>
                  <Paper sx={style.paper}>
                    <Typography variant="h6">{subject}</Typography>
                    <Stack spacing={1.5}>
                      {description?.map((value, index) => {
                        return (
                          <Stack
                            direction="row"
                            spacing={1.5}
                            alignItems="center"
                            key={index}
                            mt="16px"
                          >
                            <Typography sx={style.circle} variant="subtitle2">
                              {index + 1}
                            </Typography>
                            <Typography variant="subtitle1">{value}</Typography>
                          </Stack>
                        );
                      })}
                    </Stack>
                    <Typography
                      color="primary"
                      variant="subtitle2"
                      align="right"
                    >
                      Продолжить
                    </Typography>
                  </Paper>
                </CardActionArea>
              </Link>
            </Grid>
          );
        })}
      </Grid>
    </>
  );
};

export default TrainingLectures;
