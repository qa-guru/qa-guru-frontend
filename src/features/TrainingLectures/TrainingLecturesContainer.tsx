import React from "react";
import { useParams } from "react-router-dom";
import TrainingLectures from "./TrainingLectures";
import { useTrainingLecturesQuery } from "../../api/graphql/trainingLecture/trainingLectures";
import Spinner from "../../shared/Spinner";
import NoDataErrorMessage from "../../shared/NoDataErrorMessage";
import { useTrainingQuery } from "../../api/graphql/training/training";

const TrainingLecturesContainer: React.FC = () => {
  const { trainingId } = useParams();
  const { data: dataTrainingLectures, loading: loadingTrainingLectures } =
    useTrainingLecturesQuery({
      variables: { id: trainingId! },
    });
  const { data: dataTraining, loading: loadingTraining } = useTrainingQuery({
    variables: { id: trainingId! },
  });

  if (loadingTrainingLectures || loadingTraining) return <Spinner />;
  if (!dataTrainingLectures || !dataTraining) return <NoDataErrorMessage />;

  return (
    <TrainingLectures
      trainingId={trainingId!}
      dataTrainingLectures={dataTrainingLectures}
      dataTraining={dataTraining}
    />
  );
};

export default TrainingLecturesContainer;
