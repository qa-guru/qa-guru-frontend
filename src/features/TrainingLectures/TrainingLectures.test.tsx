import { MemoryRouter } from "react-router-dom";
import TrainingLectures from "./TrainingLectures";
import { render } from "../../test/utilsTest";
import { TrainingLecturesQuery } from "../../api/graphql/generated/graphql";

const mockDataTrainingLectures: TrainingLecturesQuery = {
  __typename: "Query",
  trainingLectures: [
    {
      __typename: "TrainingLectureDto",
      id: "string",
      number: 7,
      locking: false,
      lecture: {
        __typename: "LectureDto",
        id: "string",
        subject: "string",
        description: ["string1", "string1"],
      },
      lastLecture: {
        __typename: "LectureDto",
        id: "string",
        subject: "string",
        description: ["string1", "string1"],
      },
    },
  ],
};

const mockDataTraining: any = {
  __typename: "Query",
  training: {
    __typename: "TrainingDto",
    id: "string",
    name: "string",
    content: "string",
    techStack: null,
    tariffs: [
      {
        __typename: "TrainingTariffDto",
        id: "string",
        name: "string",
        code: "string",
        price: 7,
        homeWork: true,
        description: "string",
      },
    ],
    mentors: [
      {
        __typename: "UserDto",
        id: "string",
        firstName: "string",
        lastName: "string",
        middleName: "",
      },
    ],
  },
};

const mockTrainingId: string = "string";

describe("TrainingLectures", () => {
  it("the component is render", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <TrainingLectures
          trainingId={mockTrainingId!}
          dataTrainingLectures={mockDataTrainingLectures}
          dataTraining={mockDataTraining}
        />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
