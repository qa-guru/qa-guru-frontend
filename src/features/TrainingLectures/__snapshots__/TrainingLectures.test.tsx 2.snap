// Vitest Snapshot v1

exports[`TrainingLectures > the component is render 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <a
        href="/training/string/string"
        style="text-decoration: none;"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-6zsig3-MuiPaper-root"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
            >
              string
            </h6>
            <div
              class="css-b55nah-MuiStack-root"
            >
              <div
                class="css-1sx5dla-MuiStack-root"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 css-1h8329t-MuiTypography-root"
                >
                  1
                </h6>
                <h6
                  class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
                >
                  Git - самые основы
                </h6>
              </div>
              <div
                class="css-1sx5dla-MuiStack-root"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 css-1h8329t-MuiTypography-root"
                >
                  2
                </h6>
                <h6
                  class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
                >
                  Теория: Gradle, JUnit5, Selenide.
                </h6>
              </div>
              <div
                class="css-1sx5dla-MuiStack-root"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 css-1h8329t-MuiTypography-root"
                >
                  3
                </h6>
                <h6
                  class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
                >
                  Практика. Работаем с тренажером demoqa.com
                </h6>
              </div>
              <div
                class="css-1sx5dla-MuiStack-root"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 css-1h8329t-MuiTypography-root"
                >
                  4
                </h6>
                <h6
                  class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
                >
                  Рассмотрим подробнее возможности Selenide и CSS/Xpath-селекторы
                </h6>
              </div>
            </div>
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-alignRight css-1txwzru-MuiTypography-root"
            >
              Продолжить
            </h6>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
